### DISCUSSION TITLE
Usage of --target-triple and -march flags

### DISCUSSION THREAD

**Discussion by bjia56** [modular/modular#605] [Q&A]
`mojo build -h` reports that it supports --target-triple and -march to generate code for a target machine/architecture, however running the following on Ubuntu 22.04 x86_64 yields:
```
$ mojo build --target-triple aarch64-unknown-linux deviceinfo.mojo
mojo: error: could not construct host target info: No available targets are compatible with triple "aarch64-unknown-linux"
$ mojo build -march aarch64 deviceinfo.mojo
mojo: error: could not construct host target info: No available targets are compatible with triple "aarch64-unknown-linux"
```

Is there cross-compile support available in Mojo? Or perhaps I am misunderstanding the purpose of these two flags?

**Comment by tmke8** [comment]
What I have found out so far:

`--target-triple x86_64-unknown-linux-gnu`  works when I run it on Linux.

And

```
mojo build hello.mojo --target-triple x86_64-apple-darwin
```
compiles when I run it on Linux, but then it fails during linking:

```
/usr/bin/ld: /tmp/mojo_archive-46cdd17.a: error adding symbols: archive has no index; run ranlib to add one
collect2: error: ld returned 1 exit status
mojo: error: failed to link executable
```
